parsedatetime/__init__.py

```diff
@@ -863,7 +898,166 @@ def _quantityToReal(self, quantity):
         except KeyError:
             pass
 
-        return 0.0
+        return 0.0'''
+
+    def number_formation(self, number_words):
+        numbers = []
+        for number_word in number_words:
+            numbers.append(self.ptc.numbers[number_word])
+        if len(numbers) == 4:
+            return (numbers[0] * numbers[1]) + numbers[2] + numbers[3]
+        elif len(numbers) == 3:
+            if numbers[0] % 100 == 0:
+                return numbers[0] + numbers[1] + numbers[2]
+            else:
+                return numbers[0] * numbers[1] + numbers[2]
+        elif len(numbers) == 2:
+            if numbers[1] % 100 == 0:
+                return numbers[0] * numbers[1]
+            else:
+                return numbers[0] + numbers[1]
+        else:
+            return numbers[0]
+
+    def word_to_num(self, number_sentence):
+        if not number_sentence:
+            return 1.0
+        try:
+            return float(number_sentence.replace(',', '.'))
+        except ValueError:
+            pass
+
+        number_sentence = number_sentence.replace('-', ' ')
+        number_sentence = number_sentence.lower()  # converting input to lowercase
+        if 'ignore' in self.ptc.numbers:
+            number_sentence = number_sentence.replace(' '+self.ptc.numbers['ignore']+ ' ', ' ')
+
+        if (number_sentence.isdigit()):  # return the number if user enters a number string
+            return float(number_sentence)
+
+        clean_numbers = []
+        clean_decimal_numbers = []
+
+        # removing and, & etc.
+
+        str = number_sentence.strip() + ' '
+        m = self.ptc.CRE_MONTH.search(str)
+        if m is not None:
+            str = str.replace(m.group(), ' ')
+        m = self.ptc.CRE_WEEKDAY.search(str)
+        if m is not None:
+            str = str.replace(m.group(), ' ')
+        str = ' '+ str + ' '
+        while 1:
+            m = self.ptc.CRE_NUM.search(str)
+            if m is not None:
+                clean_numbers.append(m.group())
+                str = str[m.end():].strip()
+            else:
+                break
+
+        # Error message if the user enters invalid input!
+        if len(clean_numbers) == 0:
+            raise ValueError("No valid number words found! Please enter a valid number word (eg. two million twenty three thousand and forty nine)")
```

Comment:
Is this Error needed?

Reply:
I didn't write the errors, it was in the code originally, however even if it has no number words, an error doesn't come up in this program